src/transformStateWithClones.js

```diff
@@ -7,7 +7,48 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let cloneState = { ...state };
+  const arrayState = [];
+
+  for (const property of actions) {
+    if (property.type === 'addProperties') {
+      const addPropertiesObject = {};
+
+      for (const key in property.extraData) {
+        addPropertiesObject[key] = property.extraData[key];
+      }
+
```

Comment:
you work with `cloneState` here, do not need to create new object, just add new properties